Tests/SonarScanner.MSBuild.PreProcessor.Test/ProcessedArgsTests.cs

```diff
@@ -361,12 +390,86 @@ public void ProcArgs_OperatingSystem_AutoDetection(PlatformOS platformOS, string
+            var sut = CreateDefaultArgs(new ListPropertiesProvider([new Property(SonarProperties.UserHome, @"C:\Users\user\.sonar")]), directoryWrapper: directoryWrapper);
+            sut.UserHome.Should().BeNull();
+            sut.IsValid.Should().BeFalse();
+            logger.AssertErrorLogged(@"The provided value for 'sonar.userHome' 'C:\Users\user\.sonar' does not exists. Specify a valid directory for 'sonar.userHome'.");
+            logger.AssertNoWarningsLogged();
+        }
+
+        [TestMethod]
+        public void ProcArgs_UserHome_Default()
+        {
+            var directoryWrapper = Substitute.For<IDirectoryWrapper>();
+            var operatingSystemProvider = Substitute.For<IOperatingSystemProvider>();
+            operatingSystemProvider.GetFolderPath(Environment.SpecialFolder.UserProfile, Environment.SpecialFolderOption.None).Returns(@"C:\Users\user");
+            directoryWrapper.Exists(@"C:\Users\user\.sonar").Returns(true);
+            var sut = CreateDefaultArgs(directoryWrapper: directoryWrapper, operatingSystemProvider: operatingSystemProvider);
+            sut.UserHome.Should().Be(@"C:\Users\user\.sonar");
+            sut.IsValid.Should().BeTrue();
+            logger.AssertNoErrorsLogged();
+            logger.AssertNoWarningsLogged();
+        }
+
+        [TestMethod]
+        public void ProcArgs_UserHome_Default_CreatedOnDemand()
+        {
+            var directoryWrapper = Substitute.For<IDirectoryWrapper>();
+            var operatingSystemProvider = Substitute.For<IOperatingSystemProvider>();
+            operatingSystemProvider.GetFolderPath(Environment.SpecialFolder.UserProfile, Environment.SpecialFolderOption.None).Returns(@"C:\Users\user");
+            directoryWrapper.Exists(@"C:\Users\user\.sonar").Returns(false);
+            var sut = CreateDefaultArgs(directoryWrapper: directoryWrapper, operatingSystemProvider: operatingSystemProvider);
+            sut.UserHome.Should().Be(@"C:\Users\user\.sonar");
+            sut.IsValid.Should().BeTrue();
+            directoryWrapper.Received().CreateDirectory(@"C:\Users\user\.sonar");
+            logger.AssertNoErrorsLogged();
+            logger.AssertNoWarningsLogged();
+        }
+
+        [DataTestMethod]
+        [DynamicData(nameof(DirectoryCreateExceptions))]
+        public void ProcArgs_UserHome_Default_CreationFails(Type exceptionType)
+        {
+            var directoryWrapper = Substitute.For<IDirectoryWrapper>();
+            var operatingSystemProvider = Substitute.For<IOperatingSystemProvider>();
+            operatingSystemProvider.GetFolderPath(Environment.SpecialFolder.UserProfile, Environment.SpecialFolderOption.None).Returns(@"C:\Users\user");
+            directoryWrapper.Exists(@"C:\Users\user\.sonar").Returns(false);
+            var exception = (Exception)Activator.CreateInstance(exceptionType);
+            directoryWrapper.When(x => x.CreateDirectory(@"C:\Users\user\.sonar")).Throw(exception);
```

Comment:
Also here, the order can be improved a bit.
```suggestion
            var exception = (Exception)Activator.CreateInstance(exceptionType);
            var directoryWrapper = Substitute.For<IDirectoryWrapper>();
            directoryWrapper.Exists(@"C:\Users\user\.sonar").Returns(false);
            directoryWrapper.When(x => x.CreateDirectory(@"C:\Users\user\.sonar")).Throw(exception);
            var operatingSystemProvider = Substitute.For<IOperatingSystemProvider>();
            operatingSystemProvider.GetFolderPath(Environment.SpecialFolder.UserProfile, Environment.SpecialFolderOption.None).Returns(@"C:\Users\user");
```